### PR TITLE
Added publishing info and error logs to Che Server WebSocket endpoint

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Added publishing info and error logs to Che Server WebSocket endpoint.

In this PR there is detailed Broker logs output that is written in STDOUT and may be fetched on OpenShift console for debug purposes, and summary logs output that is published to clients.
<details>
<summary>Plugin Broker Pod Logs</summary>

```
2018/10/01 09:02:31 Broker configuration
2018/10/01 09:02:31   Push endpoint: ws://che-eclipse-che.172.19.20.145.nip.io/api/websocket
2018/10/01 09:02:31   Auth enabled: false
2018/10/01 09:02:31   Runtime ID:
2018/10/01 09:02:31     Workspace: workspacee9rue7fxrxa251f9
2018/10/01 09:02:31     Environment: default
2018/10/01 09:02:31     OwnerId: che
2018/10/01 09:02:31 Cleaning /plugins dir
2018/10/01 09:02:31 Started Plugin Broker
2018/10/01 09:02:31 List of plugins and editors to install
2018/10/01 09:02:31 - che-dummy-plugin:0.0.1 - A hello world theia plug-in wrapped into a Che Plug-in
2018/10/01 09:02:31 - che-service-plugin:0.0.1 - Che Plug-in with Theia plug-in and container definition providing a service
2018/10/01 09:02:31 - org.eclipse.che.editor.theia:1.0.0 - Eclipse Theia
2018/10/01 09:02:31 Starting plugins processing
2018/10/01 09:02:31 Stared processing plugin 'che-dummy-plugin:0.0.1'
2018/10/01 09:02:31 Downloading archive 'https://github.com/ws-skeleton/che-dummy-plugin/releases/download/untagged-8f3e198285a2f3b6b2db/che-dummy-plugin.tar.gz'  for plugin 'che-dummy-plugin:0.0.1' to '/tmp/che-plugin-broker285687574/testArchive.tar.gz'
2018/10/01 09:02:32 Untarring archive 'https://github.com/ws-skeleton/che-dummy-plugin/releases/download/untagged-8f3e198285a2f3b6b2db/che-dummy-plugin.tar.gz'  for plugin 'che-dummy-plugin:0.0.1' to '/tmp/che-plugin-broker285687574/testArchive.tar.gz'
2018/10/01 09:02:32 Resolving Che plugins for 'che-dummy-plugin:0.0.1'
2018/10/01 09:02:32 Copying dependencies for 'che-dummy-plugin:0.0.1'
2018/10/01 09:02:32 Copying file '/tmp/che-plugin-broker285687574/testArchive/hello_world_plugin.theia' to '/plugins/hello_world_plugin.theia'
2018/10/01 09:02:32 Stared processing plugin 'che-service-plugin:0.0.1'
2018/10/01 09:02:32 Downloading archive 'https://github.com/ws-skeleton/che-service-plugin/releases/download/untagged-5c4c888ff2de8ae7a5e2/che-service-plugin.tar.gz'  for plugin 'che-service-plugin:0.0.1' to '/tmp/che-plugin-broker847505277/testArchive.tar.gz'
2018/10/01 09:02:39 Untarring archive 'https://github.com/ws-skeleton/che-service-plugin/releases/download/untagged-5c4c888ff2de8ae7a5e2/che-service-plugin.tar.gz'  for plugin 'che-service-plugin:0.0.1' to '/tmp/che-plugin-broker847505277/testArchive.tar.gz'
2018/10/01 09:02:39 Resolving Che plugins for 'che-service-plugin:0.0.1'
2018/10/01 09:02:39 Copying dependencies for 'che-service-plugin:0.0.1'
2018/10/01 09:02:39 Copying file '/tmp/che-plugin-broker847505277/testArchive/che_ui_service_plugin.theia' to '/plugins/che_ui_service_plugin.theia'
2018/10/01 09:02:39 Stared processing plugin 'org.eclipse.che.editor.theia:1.0.0'
2018/10/01 09:02:39 Downloading archive 'https://github.com/ws-skeleton/che-editor-theia/releases/download/untagged-fe2d4dc402d14923b78c/che-editor-plugin.tar.gz'  for plugin 'org.eclipse.che.editor.theia:1.0.0' to '/tmp/che-plugin-broker676330168/testArchive.tar.gz'
2018/10/01 09:02:39 Untarring archive 'https://github.com/ws-skeleton/che-editor-theia/releases/download/untagged-fe2d4dc402d14923b78c/che-editor-plugin.tar.gz'  for plugin 'org.eclipse.che.editor.theia:1.0.0' to '/tmp/che-plugin-broker676330168/testArchive.tar.gz'
2018/10/01 09:02:39 Resolving Che plugins for 'org.eclipse.che.editor.theia:1.0.0'
2018/10/01 09:02:39 Copying dependencies for 'org.eclipse.che.editor.theia:1.0.0'
2018/10/01 09:02:39 All plugins have been successfully processed
```
</details>

<details>
<summary>Plugin Broker Logs Displayed on Client</summary>

Client will receive events with the following text
```
Started Plugin Broker
List of plugins and editors to install
- che-service-plugin:0.0.1 - Che Plug-in with Theia plug-in and container definition providing a service
- che-dummy-plugin:0.0.1 - A hello world theia plug-in wrapped into a Che Plug-in
- org.eclipse.che.editor.theia:1.0.0 - Eclipse Theia
Starting plugins processing
All plugins have been successfully processed
MountVolume.SetUp succeeded for volume "che-workspace-token-g2vrr" 
Container image "eclipse/che-dev:nightly" already present on machine
Created container
Started container
pulling image "wsskeleton/che-hello"
```
</details>

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/11069

### CODE REVIEW

**Comment by sleshchenko** [comment]
Depends on https://github.com/eclipse/che/pull/11422

File: broker/broker.go [hunk 1]
```diff
@@ -49,30 +52,35 @@ func Start(metas []model.PluginMeta) {
...
+		printError(err.Error())
 	}
 	bytes, err := json.Marshal(tooling)
 	if err != nil {
 		pubFailed(err.Error())
-		log.Fatalf(err.Error())
+		printError(err.Error())
 	}
+
+	printInfo("All plugins are successfully processed")
```
Comment (l0rd) [inline broker/broker.go]: All plugins have been successfully processed

File: broker/broker.go [hunk 2]
```diff
@@ -219,3 +228,33 @@ func copyDependencies(workDir string) error {
...
+
+func printError(format string, v ...interface{}) {
+	message := fmt.Sprintf(format, v...)
+	log.Fatal(message)
+}
+
+func printPlan(metas []model.PluginMeta) {
+	var buffer bytes.Buffer
+
+	buffer.WriteString("Planning to install\n")
```
Comment (l0rd) [inline broker/broker.go]: s/Planning to install/List of plugins and editors to install/

Comment (sleshchenko) [reply broker/broker.go]: Thanks. Fixed.

File: broker/broker.go (hunk 1 above)
Comment (sleshchenko) [reply broker/broker.go]: Thanks. Fixed.

File: broker/broker.go [hunk 3]
```diff
@@ -38,9 +40,10 @@ func Start(metas []model.PluginMeta) {
 	if ok, status := storage.SetStatus(model.StatusStarted); !ok {
 		m := fmt.Sprintf("Starting broker in state '%s' is not allowed", status)
 		pubFailed(m)
-		log.Fatal(m)
+		printError(m)
```
Comment (garagatyi) [inline broker/broker.go]: Would be better to use term `fatal` in the name to show ppl familiar to Golang specific naming that it is not just logging but also interrupting thread.

**Pull request merged**